k-distribution/src/main/scripts/lib/pyk/pyk/kcfg.py

```diff
@@ -65,407 +111,379 @@ def __exit__(self, exc_type, exc_value, traceback):
             return True
         return False
 
-    @staticmethod
-    def _encode(value):
-        result = None
-        if type(value) is dict:
-            result = {}
-            for (key, val) in value.items():
-                result[key] = KCFG._encode(val)
-        elif type(value) is set:
-            result = {'__kcfg_type__': 'set'}
-            val = []
-            for item in value:
-                val.append(KCFG._encode(item))
-            result['__kcfg_set_items__'] = val
-        elif isinstance(value, KInner):
-            result = value.to_dict()
-            result['__kcfg_type__'] = 'k'
-        else:
-            result = value
-        return result
+    @property
+    def nodes(self) -> Set[Node]:
+        return set(self._nodes.values())
+
+    @property
+    def init(self) -> Set[Node]:
+        return {node for node in self.nodes if self.is_init(node.id)}
+
+    @property
+    def target(self) -> Set[Node]:
+        return {node for node in self.nodes if self.is_target(node.id)}
+
+    @property
+    def stuck(self) -> Set[Node]:
+        return {node for node in self.nodes if self.is_stuck(node.id)}
+
+    @property
+    def leaves(self) -> Set[Node]:
+        return {node for node in self.nodes if self.is_leaf(node.id)}
+
+    @property
+    def covered(self) -> Set[Node]:
+        return {node for node in self.nodes if self.is_covered(node.id)}
+
+    @property
+    def uncovered(self) -> Set[Node]:
+        return {node for node in self.nodes if not self.is_covered(node.id)}
+
+    @property
+    def frontier(self) -> Set[Node]:
+        return {node for node in self._nodes.values() if self.is_frontier(node.id)}
+
+    @property
```

Comment:
Does this annotation make it memoized or something?

Reply:
It makes function `covers` accessible as an attribute, i.e. `cfg.covers`.